source/common/upstream/health_discovery_service.cc

```diff
@@ -54,12 +59,51 @@ void HdsDelegate::onReceiveInitialMetadata(Http::HeaderMapPtr&& metadata) {
   UNREFERENCED_PARAMETER(metadata);
 }
 
+void HdsDelegate::processMessage(
+    std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier>&& message) {
+  // Create HdsCluster config
+  ENVOY_LOG(debug, "Creating HdsCluster config");
```

Comment:
So, this is probably your next step, taking `message` and using the clusters and health checks specified in this proto to generate the set of clusters and corresponding health checkers.